h2/src/main/org/h2/value/ValueJson.java

```diff
@@ -0,0 +1,180 @@
+package org.h2.value;
+
+import java.io.IOException;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.Iterator;
+
+import org.h2.message.DbException;
+import org.h2.util.Utils;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+
+public class ValueJson extends Value {
+    
+    private JsonNode json;
+    private String string;
+    private int binarySize;
+    
+    private static final ObjectMapper mapper = new ObjectMapper();
+    
+    ValueJson(String str) throws IOException {
+        int memFirst = Utils.getMemoryUsed();
+        if (str == null) {
+            this.json=mapper.createObjectNode();
+            this.string = this.json.toString();
+        }
+        this.string = str.replaceAll("\n", "");
+        char c = this.string.charAt(0);
+        if ('{' == c) {
+            ObjectNode node = (ObjectNode) mapper.readTree(this.string);
+            this.json = node;
+        } else if ('[' == c) {
+            ObjectMapper mapper = new ObjectMapper();
+            ArrayNode arr = (ArrayNode) mapper.readTree(this.string);
+            this.json = arr;
+        } else {
+            TextNode stringVal = new TextNode(this.string);
+            this.json = stringVal;
+        }
+        this.binarySize = Utils.getMemoryUsed() - memFirst;
+    }
+    
+    ValueJson(JsonNode json) {
+        this.json = json;
+        this.string = json.toString().replaceAll("\n", "");
+    }
+    
+    public static ValueJson get(String str) throws IOException {
+        return new ValueJson(str);
+    }
+    
+    public static Value get(JsonNode json){
+        return new ValueJson(json);
+    }
+
+    @Override
+    public String getSQL() {
```

Comment:
Remove this method.

Reply:
Fixed [here](https://github.com/lazarevnik/h2database/commit/004d312ff1cce4325f45302e3320e8450f17eacc)